great_expectations/data_context/data_context.py

```diff
@@ -730,6 +750,29 @@ def create_expectation_suite(self, expectation_suite_name, overwrite_existing=Fa
 
         return expectation_suite
 
+    def delete_expectation_suite(self, expectation_suite_name):
+        """Delete specified expectation suite from data_context expectation store.
+
+        Args:
+            expectation_suite_name: The name of the expectation_suite to create
+
+        Returns:
+            True for Success and False for Failure.
+        """
+        expectation_suite = ExpectationSuite(expectation_suite_name=expectation_suite_name)
+        key = ExpectationSuiteIdentifier(expectation_suite_name=expectation_suite_name)
+
+        if not self._stores[self.expectations_store_name].has_key(key):
+            raise ge_exceptions.DataContextError(
+                "expectation_suite with name {} does not exist."
+            )
+        else:
+            #self._stores[self.expectations_store_name].remove_key(key)
+            del self._stores[self.expectations_store_name][key]
```

Comment:
This will not work for non in-memory stores I think; I believe that essentially the stores API needs to provide a delete method, which this should use.

Reply:
WIP at the time, resolved since